integration-libs/epd-visualization/root/util/event-listener-utils.ts

```diff
@@ -0,0 +1,57 @@
+import { Renderer2 } from '@angular/core';
+
+export interface Listener {
+  nativeElement: any;
+  eventName: string;
+  endListener: () => void;
+}
+
+export class EventListenerUtils {
```

Comment:
Just a headsup:
I can see in your feature we're using a lot of custom JS event listeners and depending on the third party code for UI interaction inside your Angular components.

This might lead to (but doesn't have to) to performance issues due to the change detection being triggered too often during UI events. I'm not saying you have those issues (you would need to determine it), but let me share the following article  "just in case" it's helpful for you in the future https://indepth.dev/posts/1434/running-event-listeners-outside-of-the-ngzone

Reply:
The main issue I was having regarding events was with Angular *not* updating when required, resulting in the animation slider handle not moving properly or not moving at all until an animation has finished playing for instance.

I have not seen any performance issues relating to event handling/angular rendering as a result of mouse/touch gestures on the animation slider (rendering frames of 2D/3D content is far more significant). It's possible that there might be a perceptible effect on really old mobile devices. If that is the case it might result in a marginally reduced frame rate when dragging the slider handle, but I think it will not be as noticeable as if the animation slider handle position is out of sync with the actual animation time position.